guillotina/db/storages/pg.py

```diff
@@ -1101,11 +1069,13 @@ async def get_conflicts(self, txn):
                 modified_oids = [k for k in txn.modified.keys()]
                 sql = self._sql.get("TXN_CONFLICTS_ON_OIDS", self._objects_table_name)
                 with watch("get_conflicts_oids"):
-                    return await self.read_conn.fetch(sql, txn._tid, modified_oids)
+                    async with self.pool.acquire() as conn:
+                        return await conn.fetch(sql, txn._tid, modified_oids)
             else:
                 sql = self._sql.get("TXN_CONFLICTS", self._objects_table_name)
                 with watch("get_conflicts"):
-                    return await self.read_conn.fetch(sql, txn._tid)
+                    async with self.pool.acquire() as conn:
+                        return await conn.fetch(sql, txn._tid)
```

Comment:
maybe it makes sense to use `conn = await txn.get_connection()` ? I guess at this point the connection is already assigned to `txn._db_conn`

Reply:
We can't. Part of the point of the read conn for these actions is it was outside the current transaction block. This operation is checking in the context outside the transaction block.

Reply:
ok, right